lib/go/Makefile

```diff
@@ -15,15 +15,37 @@ endif
 export GOPATH
 
 
+################################################################################
+##                                   DEP                                      ##
+################################################################################
+
+DEP ?= ./dep
+DEP_VER ?= 0.3.1
+DEP_BIN := dep-$$GOHOSTOS-$$GOHOSTARCH
+DEP_URL := https://github.com/golang/dep/releases/download/v$(DEP_VER)/$$DEP_BIN
+
+$(DEP):
+	GOVERSION=$$(go version | awk '{print $$4}') && \
+	GOHOSTOS=$$(echo $$GOVERSION | awk -F/ '{print $$1}') && \
+	GOHOSTARCH=$$(echo $$GOVERSION | awk -F/ '{print $$2}') && \
+	DEP_BIN="$(DEP_BIN)" && \
+	DEP_URL="$(DEP_URL)" && \
+	curl -sSLO $$DEP_URL && \
+	chmod 0755 "$$DEP_BIN" && \
+	mv -f "$$DEP_BIN" "$@"
+ifneq (./dep,$(DEP))
+dep: $(DEP)
+endif
+
+vendor: | $(DEP)
```

Comment:
Shouldn't `$(DEP)` be a hard dependency (ie., without the `|`) ?

Reply:
Ah just realized that this would download  `dep` anew every time as it's a PHONY dependency. If we let make do its job and not mark `dep` as PHONY then it shouldn't need to re-run that dependency every time.

Reply:
Again, this is to handle the case where dep is a phony dependency to point to an existing version of the binary.

Reply:
Also, it's an order only dependency because I don't care so much if the version of dep changes. It just needs to exist. That and vendor is a directory and you don't really want to compare a dir's mtime as Make does funny things with that. Order only dependencies are the best approach when depending on or being a dependency of a directory.